### PR TITLE
:source command

### PR DESCRIPTION
**Pull request by georgebrock**
`:source path` will run any Git and gitsh commands in the file at `path`. It can modify the environment of the gitsh session it's called from, much like the `source` command in Unix shells.

### CODE REVIEW

File: spec/integration/source_command_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+require 'spec_helper'
+
+describe 'The :source command' do
+  context 'a source file is given' do
+    it 'executes the commands in the sourced file' do
+      GitshRunner.interactive do |gitsh|
```
Comment (jferris) [inline spec/integration/source_command_spec.rb]: These interactive tests are nice.

File: lib/gitsh/commands/internal_command.rb [hunk 2]
```diff
@@ -155,6 +155,35 @@ def subject
...
+Runs the commands in the given file.
+TXT
+      end
+
+      def execute
+        if valid_arguments?
+          Gitsh::ScriptRunner.new(env: env).run(path)
+          true
+        else
+          env.puts_error 'usage: :source path'
```
Comment (jferris) [inline lib/gitsh/commands/internal_command.rb]: Worth extracting this usage string into a method/constant?

**Comment by jferris** [comment]
This looks good to me. I had one minor suggestion.

File: lib/gitsh/commands/internal_command.rb (hunk 2 above)
Comment (georgebrock) [inline lib/gitsh/commands/internal_command.rb]: Good idea.

File: spec/integration/source_command_spec.rb (hunk 1 above)
Comment (georgebrock) [inline spec/integration/source_command_spec.rb]: Thanks!

**Comment by georgebrock** [comment]
Thanks for the review, Joe. Mergin’

**Pull request merged**